borg/archiver.py

```diff
@@ -607,8 +607,27 @@ def do_break_lock(self, args):
         start to end, or must match from the start to just before a path separator.
         Except for the root path, paths will never end in the path separator when
         matching is attempted.  Thus, if a given pattern ends in a path separator, a
-        '*' is appended before matching is attempted.  Patterns with wildcards should
-        be quoted to protect them from shell expansion.
+        '*' is appended before matching is attempted.
+
+        Regular expressions similar to those found in Perl are also supported. Unlike
+        shell patterns regular expressions are not required to match the complete
+        path and any substring match is sufficient. It is therefore strongly
+        recommended to anchor patterns to the start ('^'), to the end ('$') or both.
+        Path separators ('\\' for Windows and '/' on other systems) in paths always
+        normalized to a forward slash ('/') before applying a pattern. The regular
+        expression syntax is described in the `Python documentation for the re module
+        <https://docs.python.org/3/library/re.html>`_.
+
+        Exclusions can be passed via command line options. The `--exclude` option
+        (`-e` for short) uses shell pattern syntax and `--exclude-regex` uses regular
+        expression syntax. When passed via a shell the patterns should be quoted to
+        protect them from expansion.
+
+        The `--exclude-from` option permits loading exclusion patterns from a text
+        file with one pattern per line. Empty lines as well as lines starting with
+        the number sign ('#') are ignored. If a line starts with '{RE}' (case
```

Comment:
can we have that as re: ?

re: is 4 key strokes, {RE} is 7..8 key strokes (german kbd) and awkward to type.

the potential, but unlikely issue that re: is part of a path/filename can be resolved in borg 1.0 when we could just always require the prefix and drop prefix-less patterns.